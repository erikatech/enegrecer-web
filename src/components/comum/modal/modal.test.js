import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Modal from './modal';

describe('Modal', () => {
  const acoes = [
    {
      nome: 'blah'
    }
  ];

  it.only('deve construir o componente corretamente', () => {
    const modal = shallow(<Modal acoes={acoes} />);
    const modalJson = toJson(modal);
    expect(modalJson).toMatchSnapshot();
  });

  it.only('deve exibir modal quando a propriedade deveExibir for true', () => {
    const modalWrapper = shallow(<Modal
      id="some"
      tituloModal="some"
      textoModal=""
      textoBotao=""
      acoes={acoes}
      deveExibir
    />);

    const found = modalWrapper.find('.modal');
    expect(found.exists()).toBeTruthy();
    expect(found).toHaveLength(1);
  });

  it.only('nÃ£o deve exibir o modal quando a propriedade deveExibir for false', () => {
    const modalWrapper = mount(<Modal
      id="some"
      tituloModal="some"
      textoModal=""
      textoBotao=""
      acoes={acoes}
      deveExibir={false}
    />);

    const found = modalWrapper.find('.modal');
    expect(found.exists()).toBeFalsy();
  });
});
