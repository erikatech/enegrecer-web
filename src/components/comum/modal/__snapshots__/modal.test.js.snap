// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal deve construir o componente corretamente 1`] = `<div />`;
